### PR TITLE
[RDY] Moved crossbows and bolts crafting over to fabrication

### PR DESCRIPTION
**Pull request by Cosmitz**

### CODE REVIEW

**Comment by Cosmitz** [comment]
No changed were done to skillbooks since i'm not sure we can lob them in with the rest, though it won't hurt to leave them bundled with the archery books. Though at a cursory reading the progression/scaling balance for crafting these is absolutely broken. I may pass them properly later. Either way, it's a fix for the current conundrum of archery-craftable items that don't use archery.

**Comment by Leland** [comment]
@Cosmitz Appreciate the work you're doing here but can you be sure to leave detailed commit messages going forward? Without a record of what a commit does and why future devs years down the road will have a difficult time figuring out why a line was changed.

**Comment by Cosmitz** [comment]
Sure, will give it a read, i have just barely finished getting Github set up after hours of frustration and i by far don't know the 'way of the land' for using it and interacting on it. ~thanks

**Comment by Leland** [comment]
Doing great so far, keep it up 👏

**Comment by Night-Pryanik** [comment]
I don't like this, as I already stated in #21626. Fabricating crossbows and bolts have nothing to do with the marksmanship. 
To be honest, I'd rather remove marksmanship skill requirement from firearms and ammo recipes.

**Comment by Cosmitz** [comment]
I agree, how i did arrow crafting for the large rebalance was like this:

image

Based on fabrication, but with a later requirement to at least be using/shooting the skill to understand what you're even wanting to craft. I can do the same for bolts for this PR, and leave the rest of the crafting requirements for guns for another time.

**Comment by kevingranade** [comment]
The reason archery is a requirement for bow and arrow crafting, and
marksmanship (previously "guns") is a requirement for ammunition crafting
is to provide an "on ramp" for exercising those skills.

I'm not saying it needs to stay the way it is, but bow and arrow crafting
in particular needs to maintain *a* way for a beginner to start practicing
the requisite skills.

The other effect of moving ammo crafting to fabrication is it provides a
VERY easy path to levelling fabrication quickly.  This isn't necessarily a
bad thing in general, but it is pretty jarring if it reaches the point of,
"craft 1,000 arrows and you're a master crafter".

IMO the link between "gun handling" and ammo reloading is slightly less
tenuous, but it's still quite a stretch.

**Comment by Cosmitz** [comment]
Fabrication will get levelled given anything we really do in the game at start and it's needed for a lot of things mid-late game.Would be neat to lower the 'i have to make 10 of these just to raise my skill' and allow a bit more organic progression in such a core skill. 

Also, moving the skills base requirement to the ammo/guns, so making bows and arrows with archery as base, will allow training archery without actually shooting archery which i always found in general to be a bit backwards. The fact that you can make it doesn't mean that you know how to use it.

The staggered way i was going to use with slaved skills would mean that you still need fabrication to even make them, but you need to shoot what you're making to learn /why/ and /what/ you should be crafting in the future. 

Makes sense, works as a ramp for gameplay, and general seems like a sound idea.

Material requirements for arrows and bows in the in-works archery PR have significantly increased from the current base, which most of it is left over from the former component-way of doing things when you made a single arrow at a time. So that will put a stop from farming arrows for components when they require a significant investment in materials.
--------
In any case, for this particular PR, i am willing to move the requirements for crafting bolts to anything you guys deem more appropriate, be it rifles or fabrication or what have you. This needs to be done in any case since atm it's a straight up relic and a bug.

**Comment by Night-Pryanik** [comment]
> The other effect of moving ammo crafting to fabrication is it provides a VERY easy path to levelling fabrication quickly. This isn't necessarily a bad thing in general, but it is pretty jarring if it reaches the point of, "craft 1,000 arrows and you're a master crafter".

Then we just need to cap training fabrication by ammo crafting to some reasonable level.

**Comment by Cosmitz** [comment]
So, are the additional requirements staying to marksmanship or moved over to rifles? Either way it makes no current sense for them to need archery.

**Comment by Night-Pryanik** [comment]
I'd prefer to leave fabrication as the only needed skill.

**Comment by Cosmitz** [comment]
I don't get that reasoning. A lot (all?) of the weapons need a core fabrication skill and then a supporting weapon skill. Why should crossbows be exempt?

**Comment by kevingranade** [comment]
The link between crossbow and quarrel crafting is tenuous, but the link between crossbow and quarrel crafting is non-existent.
The only valid options I see are for secondary skill to be either archery or nothing.

**Comment by Cosmitz** [comment]
Main in fabrication with a side-requirement in Marksmanship is completely fine and makes sense as to keeping the same progression as the rest of the weapons. It's a non-issue given marksmanship gets raised anyhow when shooting crossbows or any type of ranged weapon. A lot more sense than keeping their fabrication tied to a completely tertiary skill that not only do they gain no use out of, they'll learn to use weapons that have nothing to do with what they're using. 

This is a strictly gameplay-related matter.

**Comment by kevingranade** [comment]
There's no such thing as a "strictly gameplay-related matter".  All game features have thematic elements.
Archery or nothing, I'm fine with either, but those are the only options.

**Comment by Cosmitz** [comment]
Can i ask how is it thematically fine for the player's character to learn how to shoot a bow by crafting crossbows?

In any case, removed all secondary requirements for crafting crossbows and made the early recipes autolearn, if only to bypass the curent paradox.

**Comment by kevingranade** [comment]
Why autolearn?  That has nothing to do with the secondary skill.

**Comment by Cosmitz** [comment]
There is no secondary skill anymore, they're just based in fabrication. Autolearn is set to on for the basic starter/medium crossbow recipes as fabrication increases, with the top availible on a book-basis. Disallowing autolearn on medium-tier crossbows and bolts would mean the player would be capped to starter crossbow equipment until he finds a book.

**Comment by kevingranade** [comment]
> There is no secondary skill anymore, they're just based in fabrication.

That doesn't answer the question, why does that imply that they need to be autolearn?

> Disallowing autolearn on medium-tier crossbows and bolts would mean the player would be capped to starter crossbow equipment until he finds a book.

I might be misremembering something, but as far as I remember that's working as intended.

**Comment by Cosmitz** [comment]
I want to ask why do we want it to work like that, but that's not the point of this PR.

For right now, removed autolearn from all but the hand crossbow, added more book options so at least the rest minus top are accessibly craftable.

**Comment by kevingranade** [comment]
Please don't do unrelated things in the same PR, if you want to make these recipes more accessible in addition to changing the secondary skill, we do in fact have to put the PR on hold to discuss this other issue.

So, since that seems to be what you want to do, why do you think the hand crossbow should be autolearn?
Also why did you remove the recipes from various books?

**Comment by Cosmitz** [comment]
.. since bolts are autolearn? And that's without any contribution by this PR? 

Given the way this is advancing do you want to remove any ability of a starting character to even make any sort of crossbow related weapons until they found either the items themselves out in the wild (which most aren't in spawn groups anyway) or the books for them? 

Since i just did that in the last commits. I reverted any changes to book requirements, and accomplished what this PR aimed to do, finish the proper removal of crossbows out of archery.

I just hope someone that really cares for the crossbows as weapon systems does a proper pass over them at some point since they are a horrid mess right now.

File: data/json/recipes/recipe_weapon.json [hunk 1]
```diff
@@ -1001,13 +1001,11 @@
...
   "category": "CC_WEAPON",
   "subcategory": "CSC_WEAPON_RANGED",
-  "skill_used": "mechanics",
-  "skills_required": [ "archery", 3 ],
+  "skill_used": "fabrication",
   "difficulty": 3,
   "time": 30000,
   "reversible": true,
   "decomp_learn": 3,
-  "book_learn" : [[ "recipe_bows", 3 ] , [ "manual_archery", 5] , [ "book_archery", 4]],
```
Comment (kevingranade) [inline data/json/recipes/recipe_weapon.json]: You missed one.

File: data/json/recipes/recipe_weapon.json [hunk 2]
```diff
@@ -1036,9 +1034,8 @@
   "result": "crossbow",
   "category": "CC_WEAPON",
   "subcategory": "CSC_WEAPON_RANGED",
-  "skill_used": "mechanics",
-  "skills_required": [ "archery", 3 ],
-  "difficulty": 3,
+  "skill_used": "fabrication",
+  "difficulty": 2,
```
Comment (kevingranade) [inline data/json/recipes/recipe_weapon.json]: Why the change in difficulty?

File: data/json/recipes/recipe_weapon.json [hunk 3]
```diff
@@ -4671,9 +4665,8 @@
   "result": "rep_crossbow",
   "category": "CC_WEAPON",
   "subcategory": "CSC_WEAPON_RANGED",
-  "skill_used": "mechanics",
-  "skills_required": [ "archery", 3 ],
-  "difficulty": 3,
+  "skill_used": "fabrication",
+  "difficulty": 4,
```
Comment (kevingranade) [inline data/json/recipes/recipe_weapon.json]: Why the change in difficulty?

**Comment by kevingranade** [comment]
> .. since bolts are autolearn? And that's without any contribution by this PR?

What does that have to do with crossbows?  One is a straightforward shaft of wood with a head and fletchings, the other is a complex elastic projectile launcher.  The difficulty of crafting them is incomparable.

Since you ask, I'm not particularly interested in starting characters being able to craft crossbows, they're quite complex and require a lot of precision, and it's not at all reasonable for a survivor to spontaneously figure out how to make one, that's why they weren't set to autolearn in the first place.

**Comment by Cosmitz** [comment]
Fixed lingering difficulty changes and readded books to the hand crossbow.
